nhs/utils/logging.py

```diff
@@ -49,32 +51,43 @@ def config_logger():
     )
 
     logging.basicConfig(handlers=[__InterceptHandler()], level=0, force=True)
-    warnings.showwarning = lambda msg, *args, **kwargs: logger.warning(msg)
 
-
-def log_entry_exit(*, entry=True, exit=True, level="DEBUG"):
+    def custom_showwarning(
+        message: Union[Warning, str],
+        category: Type[Warning],
+        filename: str,
+        lineno: int,
+        file: Optional[TextIO] = None,
+        line: Optional[str] = None
+    ) -> None:
+        logger.warning(f"{category.__name__}: {str(message)}")
+
+    warnings.showwarning = custom_showwarning
```

Comment:
Custom warning required to make optional warning.
Otherwise pylance gets mad about how the parameters are unused or partially unused or unaccessed.

Reply:
also btw if the pylance/pyright warning is nonsensical, e.g. because it can't deduce the type, you can use `#type ignore`

e.g. when you run a data through a list of functions pyright can't tell what type the final data will be despite it only needing to see what the last function returns >:(

```python
data = 3
for func in [str, tuple, list]:
    data = func(data)

data   # pyright is stoopid and don't know data is a 'list'
```